add GitHub templates

joshgav opened the pull request:
Updates and adds PR and issue templates. Multiple issue templates provide experience described [here](https://help.github.com/articles/creating-issue-templates-for-your-repository/); I determined GitHub's expected format by trying it with the described UI :).

Targeting `master` cause these templates only work when on the master branch.

joshgav commented:
closes #1234

azuresdkci commented:
Can one of the admins verify this patch?

jhendrixMSFT commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,21 @@
+
+Describe your environment, including:
+
+* Azure Service (e.g. compute): 
```
Can we change this to package?  Sometimes it's not always clear which package is being used especially when there are multiple API versions.

joshgav replied:
What do you think of replacing current line with `SDK Package (e.g. "/services/compute/mgmt/2018-04-01/compute"):`?

vladbarosan commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,12 +1,14 @@
+- [] The purpose of the PR is explained in this or a referenced issue.
+- [] The PR does not update automatically generated files.
+   - These files are managed by the codegen framework at [Azure/autorest.go][].
+- [] The PR targets the `dev` branch.
```
do we want the non gen changes to go into dev or in "latest " ?

joshgav replied:
probably latest actually. should we get rid of `dev`? Is there a reason to separate human contributions from machine ones?

joshgav replied:
changed in latest push

vladbarosan commented on .github/ISSUE_TEMPLATE/work_item.md:
```diff
@@ -0,0 +1,8 @@
+---
+name: Work Item
```
I guess this one is more for internal WI ?

vladbarosan replied:
was wondering if we really needed, if we can incorporate it in the feature/bug or maybe rename it as infra or other

joshgav replied:
i'm thinking of removing it. these are templates after all, we can just erase them and write what we want :)

joshgav commented:
Made major changes in this edit pass. Removed "Work Item" cause we can always replace the template if needed. Added a "service package request" since that's pretty common and would help to track and encourage.

Targeting latest instead of master per our own instructions :). I think we should have everything land in `latest` and have a way to backport to master if needed in a current release.

jhendrixMSFT commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,29 @@
+Thank you for opening a bug report. For faster processing, please include:
+-->
+
+* import path of package in question, e.g. `.../services/compute/mgmt/2018-06-01/mgmt`
```
I think the trailing ```mgmt``` should be ```compute```.

joshgav replied:
fixed

jhendrixMSFT commented on .github/ISSUE_TEMPLATE/bug_report.md:
```diff
@@ -0,0 +1,29 @@
+-->
+
+* import path of package in question, e.g. `.../services/compute/mgmt/2018-06-01/mgmt`
+* SDK version, e.g. `master`, `latest`, `18.1.0`
```
The branch names might be too vague, we should probably ask for commit hashes if they aren't pinned to a semver.

joshgav replied:
Makes sense, but I'd like to provide succinct instructions on how to find the revision. I added a hint about using `dep status`, is there something else we'd recommend, like running a git command in the local source dir?

marstr commented on .github/PULL_REQUEST_TEMPLATE.md:
```diff
@@ -1,12 +1,16 @@
+- [ ] The PR targets the `latest` branch.
+- [ ] Tests are included and/or updated for code changes.
+- [ ] Updates to [CHANGELOG.md][] are included.
+- [ ] Apache v2 license headers are included in each file.
```
[off-topic] what would it take for us to change our license to MIT?

joshgav replied:
Lots of time with the legal team :). Let's discuss motivations for this offline.

joshgav commented:
landed in latest, will open another PR to backport to master
